overlord/snapshotstate/snapshotstate.go

```diff
@@ -333,6 +333,19 @@ func Import(ctx context.Context, st *state.State, r io.Reader) (setID uint64, sn
 	}
 	snapNames, err = backendImport(ctx, setID, r)
 	if err != nil {
+		if dupErr, ok := err.(backend.DuplicatedSnapshotImportError); ok {
```

Comment:
it's very unlikely but given we don't hold any kind of lock there's a chance that the snapshot is getting forgotten around here, before we get a chance to take the state lock.

It might be that easiest solution is really to have some kind of snapshot manager local lock to use around the listing of what is on disk and until the end of initialisation of some operations...